hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestChecker.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.workflows.ingest;
+
+import com.hedera.node.app.spi.key.HederaKey;
+import com.hedera.services.store.models.Account;
+import com.hederahashgraph.api.proto.java.SignatureMap;
+import com.hederahashgraph.api.proto.java.SignedTransaction;
+import com.hederahashgraph.api.proto.java.Transaction;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import java.util.List;
+import javax.annotation.Nonnull;
+
+/**
+ * Encapsulates the workflow related to transaction ingestion. Given a Transaction, parses,
+ * validates, and submits to the platform.
+ */
+public interface IngestChecker {
+
+    /**
+     * Validates a {@link Transaction}
+     *
+     * @param tx the {@code Transaction}
+     * @throws PreCheckException if validation fails
+     * @throws NullPointerException if {@code tx} is {@code null}
+     */
+    void checkTransaction(@Nonnull Transaction tx) throws PreCheckException;
+
+    /**
+     * Validates a {@link SignedTransaction}
+     *
+     * @param tx the {@code SignedTransaction} to check
+     * @throws PreCheckException if validation fails
+     * @throws NullPointerException if {@code tx} is {@code null}
+     */
+    void checkSignedTransaction(@Nonnull SignedTransaction tx) throws PreCheckException;
+
+    /**
+     * Validates a {@link TransactionBody} with the paying {@link Account}
+     *
+     * @param txBody the {@code TransactionBody} to check
+     * @param account the paying {@code Account}
+     * @throws PreCheckException if validation fails
+     * @throws NullPointerException if any of the parameters is {@code null}
+     */
+    void checkTransactionBody(@Nonnull TransactionBody txBody, @Nonnull Account account)
+            throws PreCheckException;
+
+    /**
+     * Validates a signature.
+     *
+     * @param platformTx the signed bytes to check
+     * @param signatureMap the {@link SignatureMap} with all signatures
+     * @param reqKeys a list of required {@link HederaKey}s
+     * @throws PreCheckException if validation fails
+     * @throws NullPointerException if any of the parameters is {@code null}
+     */
+    void verifySignatures(
```

Comment:
Since this is ingest we would be verifying only payer signature. Better to rename this method to `verifyPayerSignature` ?

Reply:
Yes, I think this requires some cleanup. There are checks, which are shared between ingest and pre-handle and other checks are unique to one or the other. This is not properly reflected in the interfaces yet. I plan to clean this up once I start to implement the ingest workflow.